Fixes regional forms not showing in the dexnav

miriamlefae opened the pull request:
DexNav now shows regional forms of pokemon whose other forms are present in the same route.

## Description
~~An array of regional forms was added in `pokemon.c`~~. Two functions were added. One to check if a pokemon is in the array and another to check if it has a regional form (check array for matching dex num). The dexnav was modified to return false if the pokemon is or has a regional form and it's `speciesId` is not in the array.

---
**EDIT:**
Made the following changes in response to review:
- Changed the`IsRegionalForm` and `HasRegionalForm` helper functions to not use the `sRegionalFormSpecies` array.
- Instead the `isAlolanForm`, `isGalarianForm`, `isHisuianForm,` and `isPaldeanForm` members in the `SpeciesInfo` struct are used.
- `HasRegionalForm` now uses the pokemon's `formSpeciesIdTable` to check if it has regional forms.
---

## Images
<img width="240" height="160" alt="pokeemerald-8" src="https://github.com/user-attachments/assets/26c7c45f-c37f-4458-8837-918b26ec526a" />


## Feature(s) this PR does NOT handle:
This does not handle forms other than regional forms if they are to be handled.


## **Discord contact info**
kildemal

miriamlefae committed:
fix: show regional forms in dexnav

- DexNav now shows regional forms of pokemon whose other forms are present in the same route.
- Adds array of regional forms
- Adds functions for checking regional forms

iriv24 commented on src/pokemon.c:
```diff
@@ -7311,6 +7465,52 @@ bool32 SpeciesHasGenderDifferences(u16 species)
 
+bool32 IsRegionalForm(u16 speciesId)
+{
+    if (!P_REGIONAL_FORMS)
```
i think this function can be simplified by getting the species data from the species and checking for these flags
<img width="645" height="438" alt="image" src="https://github.com/user-attachments/assets/32ea6512-e950-49a3-8196-170d899c867a" />

miriamlefae replied:
In hindsight I should've searched for more than isRegionalForm before writing that whole array. Changing

iriv24 replied:
it looked like the array is still used for the HasRegionalForm function though right?

miriamlefae replied:
It is, yes. Could make a smaller array for it though if the performance is a concern.

iriv24 replied:
nah its fine, i dont think performance is an issue unless you notice the dexnav like lagging when you first open it haha

miriamlefae replied:
Simplified both functions to not use the array.

miriamlefae committed:
refactor: regional form funcs work without array

- Changes the IsRegionalForm and HasRegionalForm helper functions to not use the sRegionalFormSpecies array.
- Instead the isAlolanForm, isGalarianForm, isHisuianForm, and isPaldeanForm members in the SpeciesInfo struct are used.
- HasRegionalForm now uses the pokemon's formSpeciesIdTable to check if it has regional forms.

miriamlefae committed:
refactor: remove unused region form array
